include/arch/riscv64/stdint.h

```diff
@@ -16,51 +16,133 @@
 #ifndef _LIBPHOENIX_ARCH_RISCV64_STDINT_H_
 #define _LIBPHOENIX_ARCH_RISCV64_STDINT_H_
 
+/* The following sections refer to ISO/IEC 9899:1999 */
+
+/* 7.18.1.1 Exact-width integer types */
+
+typedef signed char int8_t;
+typedef short int16_t;
+typedef int int32_t;
+typedef long int64_t;
+
 typedef unsigned char uint8_t;
 typedef unsigned short uint16_t;
 typedef unsigned int uint32_t;
 typedef unsigned long uint64_t;
 
-typedef uint8_t uint_fast8_t;
-typedef uint16_t uint_fast16_t;
-typedef uint32_t uint_fast32_t;
-typedef uint64_t uint_fast64_t;
+/* 7.18.1.2 Minimum-width integer types */
+
+typedef int8_t int_least8_t;
+typedef int16_t int_least16_t;
+typedef int32_t int_least32_t;
+typedef int64_t int_least64_t;
 
 typedef uint8_t uint_least8_t;
+typedef uint16_t uint_least16_t;
+typedef uint32_t uint_least32_t;
+typedef uint64_t uint_least64_t;
 
-typedef signed char int8_t;
-typedef short int16_t;
-typedef int int32_t;
-typedef long int64_t;
+/* 7.18.1.3 Fastest minimum-width integer types */
 
 typedef int8_t int_fast8_t;
-typedef int16_t int_fast16_t;
-typedef int32_t int_fast32_t;
+typedef int64_t int_fast16_t;
+typedef int64_t int_fast32_t;
```

Comment:
Isn't `int` supposed to be the fastest type by default? It might be a reason why even though this is 64 bit arch, `int` is still 32 bits wide only.

Reply:
I have checked what rv64gc gcc 10.2.0 and rv64gc clang 13.0.0 report. For (u)int_fast8_t it was 8 bits, and 64 bits for the wider fast types.